tasks/dist.js

```diff
@@ -0,0 +1,106 @@
+module.exports = function(grunt) {
+
+    // location of our temporary working copy we use to build tags
+    var tempCheckout = 'temp/checkout';
+
+    // register a few promise related convenience functions
+    var exec = require('./utils')(grunt).exec;
```

Comment:
I would still need to create a chained promise to keep the logging aspect. Is replacing a sync'ed call worth the dependency?